pkg/workflows/upgrade.go

```diff
@@ -513,10 +518,13 @@ func (s *upgradeWorkloadClusterTask) Run(ctx context.Context, commandContext *ta
 	err := commandContext.ClusterManager.UpgradeCluster(ctx, commandContext.ManagementCluster, commandContext.WorkloadCluster, commandContext.ClusterSpec, commandContext.Provider)
 	if err != nil {
 		commandContext.SetError(err)
-		logger.Info("Backing up management components from bootstrap cluster")
-		err := commandContext.ClusterManager.BackupCAPI(ctx, commandContext.ManagementCluster, commandContext.ManagementClusterStateDir, commandContext.WorkloadCluster.Name)
-		if err != nil {
-			logger.Info("Bootstrap management component backup failed, use existing workload cluster backup", "error", err)
+		// Take backup of bootstrap cluster capi components
+		if commandContext.BootstrapCluster != nil {
+			logger.Info("Backing up management components from bootstrap cluster")
+			err := commandContext.ClusterManager.BackupCAPIWaitForInfrastructure(ctx, commandContext.BootstrapCluster, fmt.Sprintf("bootstrap-%s", commandContext.ManagementClusterStateDir), "")
```

Comment:
We technically only need the cluster being upgraded here right, which is the cluster being upgraded? Since we are already attempting to back up all before, but if you think it's better to get more here as well that's fine with me.

Reply:
From my understanding we just want to backup the CAPI components on the bootstrap cluster. Are you saying we should have a `--filter-cluster` parameter here with the workload cluster name?

Reply:
Yes we could keep it with the parameter, but it doesn't make a huge difference to me.

Reply:
Gotcha, I'll go ahead and keep the parameter in for this.